schema.ts

```diff
@@ -1280,6 +1292,18 @@ export enum PrincipalType {
 export interface UserPrincipal {
   type: PrincipalType.User;
   userId: string | number;
+  userIdType?: UserIdType;
```

Comment:
If the userId type is anyone or domain, is userId filled in?

Reply:
No that is a different type that does not require userId

Reply:
Oh whoops.  I was very confused.  I didn't realize I conflated the UserIdType w/PrincipalType.  You should document the default (presumably id?) if this is undefined.  Or perhaps better, ensure it is defined?

Reply:
With the type I was not sure the best answer (for backwards compatibility w/ packs that already provide permissions. We can require it since those packs are all internal)